src/lib/arch/aarch64/ExceptionHandler.cc

```diff
@@ -13,462 +10,98 @@ namespace simeng {
 namespace arch {
 namespace aarch64 {
 
-ExceptionHandler::ExceptionHandler(
-    const std::shared_ptr<simeng::Instruction>& instruction, const Core& core,
-    MemoryInterface& memory, std::shared_ptr<OS::SyscallHandler> sysHandler)
-    : instruction_(*static_cast<Instruction*>(instruction.get())),
-      core(core),
-      memory_(memory),
-      sysHandler_(sysHandler) {
-  resumeHandling_ = [this]() { return init(); };
+ExceptionHandler::ExceptionHandler(const Core& core) : core_(core) {}
+
+bool ExceptionHandler::tick() {
+  // If the syscall handler has been envoked, conclude the syscall only once the
+  // result has been returned
+  if (envokingSycallHandler_) {
+    if (!syscallReturned_) return false;
+    return concludeSyscall();
+  }
+
+  // If an instruction ahs been resgitered with an exception, begin its
+  // processing
+  if (instruction_ != nullptr) return handleException();
+
+  // Return false as a default case
+  return false;
+}
+
+void ExceptionHandler::registerException(
+    std::shared_ptr<simeng::Instruction> instruction) {
+  // TODO: look into dynamic_pointer_cast for safer casting
+  instruction_ = std::static_pointer_cast<Instruction>(instruction);
```

Comment:
A static_cast should be fine here because we know at compile time which instruction type will be passed in i.e aarch64::Instruction will always be passed to a aarch64::ExceptionHandler. A dynamic cast would be required if we didnt know the type of instruction passed at compile time i.e if we didnt know whether aarch64::ExceptionHandler would receive riscv::Instruction or aarch64::Instruction. Since we can guarantee that aarch64::ExceptionHandler always takes in aarch64::Instruction, there is no need for a dynamic_cast. 
Essentially this is the equivalent of "trust me bro" to the compiler, but in this case we are trustworthy always.

Also I would suggest rewriting it is:
```
instruction_ = std::static_pointer_cast<aarch64::Instruction>(...)
```
Just to avoid any confusion, I thought `simeng::Instruction` was being converted to some base class `Instruction`, took me a while to realise we are actually in the aarch64 namespace

Reply:
From a quick research and your comments, think we can trust static cast here. Made cast type change also